src/client/baseClient.ts

```diff
@@ -116,10 +128,58 @@ export class BaseClient {
       const domain = (baseMetrics.domain = await this.resolveDomain(
         this.config.uiCsdsServiceName
       ));
-      const resp = await this.performGetLambdasRequest(
-        lambdaRequestData,
+
+      const isV2 = this.isV2Domain(domain);
+
+      const resp = isV2
+        ? await this.performGetFunctionsRequest(
+            {
+              state:
+                typeof lambdaRequestData.state === 'string'
+                  ? [lambdaRequestData.state]
+                  : lambdaRequestData.state,
+              skillId: lambdaRequestData.skillId,
+              eventId: lambdaRequestData.eventId,
+            },
+            domain
+          )
+        : await this.performGetLambdasRequest(lambdaRequestData, domain);
+
+      const successMetric = this.enhanceBaseMetrics(baseMetrics, {
+        requestDurationInMillis: watch.read(),
+      });
+      this.tooling.metricCollector?.onGetLambdas(successMetric);
+      return resp;
+    } catch (error) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const statusCode = ((error as VError)?.cause as any)?.jse_cause?.jse_info
+        ?.response;
+      const failureMetric = this.enhanceBaseMetrics(baseMetrics, {
+        requestDurationInMillis: watch.read(),
+        statusCode,
+        error,
+      });
+
+      this.tooling.metricCollector?.onGetLambdas(failureMetric);
+      throw error;
+    } finally {
+      watch.stop();
+    }
+  }
+
+  async getFunctions(functionRequestData: FunctionRequest): Promise<Response> {
+    const baseMetrics = this.collectBaseMetricsFrom(functionRequestData); // FIXME
```

Comment:
Is this fixed or do we still need to fix it ? And if so what is broken. Else remove it

```suggestion
    const baseMetrics = this.collectBaseMetricsFrom(functionRequestData); // FIXME
```

Reply:
remove the comment since there was not issues